spec/system/grant_submissions/grant_submission_responses_spec.rb

```diff
@@ -8,137 +8,342 @@ def successful_application_message
             find_field('Short Text Question', with:'').set(Faker::Lorem.sentence)
-            find_field(number_question.text, with: '').set('123.00aaa')
+            find_field('Number Question', with:'').set(Faker::Number.number(digits: 10))
             find_field('Long Text Question', with:'').set(Faker::Lorem.paragraph_by_chars(number: 1000))
+
             click_button 'Submit'
             expect(page).not_to have_content successful_application_message
-            expect(page).to have_content "Response to '#{number_question.text}' must be a number."
           end
         end
 
-        scenario 'requires a file_upload response' do
-          file_upload_question.update_attribute(:is_mandatory, true)
-          find_field('Short Text Question', with:'').set(Faker::Lorem.sentence)
-          find_field('Number Question', with:'').set(Faker::Number.number(digits: 10))
-          find_field('Long Text Question', with:'').set(Faker::Lorem.paragraph_by_chars(number: 1000))
+        context 'combined questions' do
+          context 'file_upload' do
+            context 'with another invalid response' do
+              before(:each) do
+                find_field('Number Question', with:'').set('Number')
+                page.attach_file(file_upload_question.text, Rails.root.join('spec', 'support', 'file_upload', 'text_file.pdf'))
+                click_button 'Submit'
+              end
 
-          click_button 'Submit'
-          expect(page).not_to have_content successful_application_message
+              scenario 'do not receive success message on field with wrong type of response' do
+                expect(page).not_to have_content successful_application_message
+                expect(page).to have_link 'text_file.pdf'
+              end
+
+              scenario 'submits file successfully when file when validation error is corrected' do
+                find_field('Number Question', with: 'Number').set(10)
+                click_button 'Submit'
+                expect(page).to have_content successful_application_message
+                expect(GrantSubmission::Response.find_by(question: file_upload_question).document.attached?).to be true
+              end
+            end
+          end
         end
       end
 
-      context 'combined questions' do
-        context 'file_upload' do
-          context 'with another invalid response' do
+      context 'states' do
+        let(:file_upload_question) {@grant.questions.find_by(response_type: 'file_upload')}
+
+        context 'required questions' do
+          context 'on create' do
+            context 'saving as draft' do
+              context 'requires response mandatory short_text question' do
+                scenario 'no answer for required short text' do
+                  short_text_question = @grant.questions.where(response_type: 'short_text').first
+                  short_text_question.update_attribute(:is_mandatory, true)
```

Comment:
These two lines could be combined into one without assigning a variable. 

Looks like short_text_question.is_mandatory is set in all subsequent scenarios. Could it be included in a before(:each)?